pkg/puller/puller.go

```diff
@@ -0,0 +1,76 @@
+package puller
+
+import (
+	"log"
+)
+
+var (
+	p *Puller
+)
+
+type Puller struct {
+	// TODO: Should this be a syncmap?
+	ChannelMap map[string]Channel
+}
+
+type Channel struct {
+	EventChannel chan EventWrapper
+}
+
+func (*Puller) AddModel(modelName string, numWorkers int) Channel {
+	// TODO: Figure out the appropriate buffer-size for this
+	eventChannel := make(chan EventWrapper)
+	channel := Channel{
+		EventChannel: eventChannel,
+	}
+	// TODO: Figure out parallelism
+	for workers := 1; workers <= numWorkers; workers++ {
+		go worker(workers, modelName, channel.EventChannel)
```

Comment:
```suggestion
		go modelProcessor(workers, modelName, channel.EventChannel)
```
How about rename worker to modelProcessor so it is more specific?